pkg/retention/retention.go

```diff
@@ -196,7 +195,7 @@ func GetRetentionTimeMs(retentionHours int, currTime time.Time) uint64 {
 func deleteSegmentsFromEmptyPqMetaFiles(segmentsToDelete map[string]*structs.SegMeta) {
 	for _, segmetaEntry := range segmentsToDelete {
 		for pqid := range segmetaEntry.AllPQIDs {
-			pqsmeta.DeleteSegmentFromPqid(pqid, segmetaEntry.SegmentKey)
+			go writer.RemoveSegmentFromEmptyPqmeta(pqid, segmetaEntry.SegmentKey)
```

Comment:
we don't want a go routine here, as it will spawn 1000*NumSegsToDelete gorutines. 
It is ok to have this as a blocking call.